packages/overseer-rs/src/criteria/block_timeliness.rs

```diff
@@ -0,0 +1,152 @@
+use std::time::SystemTime;
+
+use anyhow::Result;
+
+use super::{BlacklistCriterion, EvaluationContext};
+use crate::types::Violation;
+
+/// Checks whether the latest block arrival exceeds the configured time threshold.
+#[derive(Debug, Default)]
+pub struct BlockTimelinessCriterion;
+
+impl BlockTimelinessCriterion {
+    /// Creates a new block timeliness criterion instance.
+    pub fn new() -> Self {
+        Self
+    }
+}
+
+#[async_trait::async_trait]
+impl BlacklistCriterion for BlockTimelinessCriterion {
+    /// Flags a violation when no valid block is observed within the allowed delay.
+    async fn evaluate(&self, ctx: &EvaluationContext<'_>) -> Result<Option<Violation>> {
+        let last_block_time = ctx.observation.latest_block.timestamp;
+        let now = SystemTime::now();
+        let Ok(delay) = now.duration_since(last_block_time) else {
```

Comment:
I think that error should be propagated here:
`let delay = now.duration_since(last_block_time)?;`
other way, there is no reason to return `Result` from the function, because it always will be `Ok`.